tests/temp_tests/test_colocatedJobGroup.py

```diff
@@ -10,8 +10,13 @@
 app_3 = Application("app_3", "python", run_settings=LaunchSettings("slurm"))
 
 
+class MockJob(BaseJob):
```

Comment:
why did you switch out BaseJob for MockJob?

Reply:
We need to provide a concrete implementation of the `BaseJob` abstract class. Otherwise we cannot instance the class we are testing!